bbb/services.py

```diff
@@ -271,30 +271,34 @@ def _handle_taskcluster_exceptions(self, t, exc):
         status_code = exc.superExc.response.status_code
 
         if status_code == 409:
-            log.info("Deadline exceeded for task %s run %s, cancelling it",
+            log.warn("Deadline exceeded for task %s run %s, cancelling it",
                      t.taskId, t.runId)
             branch = self.buildbot_db.getBranch(t.buildrequestId).split("/")[-1]
             try:
                 for id_ in self.buildbot_db.getBuildIds(t.buildrequestId):
                         self.selfserve.cancelBuild(branch, id_)
-            except RequestException:
-                log.exception("Failed to cancel task %s run %s build request ID %s",
-                              t.taskId, t.runId, t.buildrequestId)
-            else:
                 # delete from the DB only if all cancel requests pass
                 self.bbb_db.deleteBuildRequest(t.buildrequestId)
 
+            except RequestException:
+                log.exception(
+                    "Failed to cancel task %s run %s build request ID %s",
+                    t.taskId, t.runId, t.buildrequestId)
+
         elif status_code == 404:
+            # Expired tasks are removed from the TC DB
             log.warn("Cannot find task %s run %s in TC, removing it", t.taskId,
                      t.runId)
             self.bbb_db.deleteBuildRequest(t.buildrequestId)
+
         elif status_code == 403:
+            # Bug 1270785. Claiming a completed task returns 403.
             log.warn("Cannot modify task %s run %s in TC, removing it",
                      t.taskId, t.runId)
             self.bbb_db.deleteBuildRequest(t.buildrequestId)
         else:
-            log.exception("Unhandled TC status code %s for task %s run %s",
-                          status_code, t.taskId, t.runId)
+            log.warn("Unhandled TC status code %s for task %s run %s",
+                     status_code, t.taskId, t.runId)
```

Comment:
`log.warn()` adds only a single line to already printed traceback. I still want to see the tracebacks until we localize all edge cases.